src/common/primitive_iface.cpp

```diff
@@ -97,11 +97,23 @@ status_t primitive_execute(
 
     if (get_verbose(verbose_t::exec_profile,
                 prim_kind2_comp_kind(primitive_iface->pd()->impl()->kind()))) {
+
+#ifdef EXPERIMENTAL_ASYNC_VERBOSE
+        // tracks the execution times for the primitive asynchronously by
+        // directly querying the event times for the stream.
+        // gets rid of the synchronization overheads on the host device
+        double start_ms, duration_ms;
+        CHECK(stream->register_async_tracker(&start_ms, &duration_ms));
```

Comment:
start_ms and duration_ms are on the stack, I am surprised the callback function does not segfault since those are out of scope when the callback function is running.